docs/docs/tutorials/rag/langgraph_agentic_rag.ipynb

```diff
@@ -126,315 +225,561 @@
+    "                \"role\": \"user\",\n",
+    "                \"content\": \"What does Lilian Weng say about types of reward hacking?\",\n",
+    "            },\n",
+    "            {\n",
+    "                \"role\": \"assistant\",\n",
+    "                \"content\": \"\",\n",
+    "                \"tool_calls\": [\n",
+    "                    {\n",
+    "                        \"id\": \"1\",\n",
+    "                        \"name\": \"retrieve_blog_posts\",\n",
+    "                        \"args\": {\"query\": \"types of reward hacking\"},\n",
+    "                    }\n",
+    "                ],\n",
+    "            },\n",
+    "            {\"role\": \"tool\", \"content\": \"meow\", \"tool_call_id\": \"1\"},\n",
+    "        ]\n",
+    "    )\n",
+    "}\n",
+    "grade_documents(input)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ba25a787-004e-4e6c-8f22-62087af7da1d",
+   "metadata": {},
+   "source": [
+    "We can also confirm that the relevant documents are classified as such:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "8bd5d24b-f05e-47c4-b954-267ab514a090",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'generate_answer'"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "input = {\n",
+    "    \"messages\": convert_to_messages(\n",
+    "        [\n",
+    "            {\n",
+    "                \"role\": \"user\",\n",
+    "                \"content\": \"What does Lilian Weng say about types of reward hacking?\",\n",
+    "            },\n",
+    "            {\n",
+    "                \"role\": \"assistant\",\n",
+    "                \"content\": \"\",\n",
+    "                \"tool_calls\": [\n",
+    "                    {\n",
+    "                        \"id\": \"1\",\n",
+    "                        \"name\": \"retrieve_blog_posts\",\n",
+    "                        \"args\": {\"query\": \"types of reward hacking\"},\n",
+    "                    }\n",
+    "                ],\n",
+    "            },\n",
+    "            {\n",
+    "                \"role\": \"tool\",\n",
+    "                \"content\": \"reward hacking can be categorized into two types: environment or goal misspecification, and reward tampering\",\n",
+    "                \"tool_call_id\": \"1\",\n",
+    "            },\n",
+    "        ]\n",
+    "    )\n",
+    "}\n",
+    "grade_documents(input)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4c337d45-bc35-4e88-8b27-ebc6631f9fed",
+   "metadata": {},
+   "source": [
+    "### Rewrite question"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e8d67fdf-0b77-4b90-97f2-63a571c3e496",
+   "metadata": {},
+   "source": [
+    "If the documents are classified as irrelevant, we will call `rewrite_question` node to try and improve the original user question:"
```

Comment:
Are we expecting some docs to be irrelevant? If so, we probably want to make this more definitive like "Documents can be returned as `irrelevant`, which indicates a need to improve the original user question. To do so, we will call..." (or something along those lines)